### DISCUSSION TITLE
Windows: Debug of OpenSC

### DISCUSSION THREAD

**Discussion by crazydev0206** [OpenSC/OpenSC#3197] [Q&A] [answered]
Hello everyone
After downloading the OpenSC project, I tried to debug it in Visual Studio 2022. 
However, I was unsure how to set up the project.
If anyone has experience, please let me know.
Thank you

**Comment by frankmorgner** [comment, marked as answer]
Guidance can be found on the wiki:
https://github.com/OpenSC/OpenSC/wiki/Compiling-on-Windows

This is how we compile OpenSC via Visual Studio in the CI environment:
https://github.com/OpenSC/OpenSC/blob/master/.appveyor.yml

Once you have compiled OpenSC, you'll see the pdb files which allow you inspection with VS of any program that loads the OpenSC minidriver or PKCS#11 module.

**Reply by crazydev0206** [reply]
Hi Frank.
I appreciate your help.
Hope to discuss more in PIV smart card.

https://github.com/OpenSC/OpenSC/discussions/3204

Thank you

**Comment by crazydev0206** [comment]
Hi Frank

I appreciate your help.

Thomas Wang
